[peienlim] iP

peienlim opened the pull request:
## **Cookie 🍪**

> "The way to get started is to quit talking and begin doing." - Walt Disney [source](https://dansilvestre.com/productivity-quotes/)

Free your mind of all the tasks you have, let Cookie keep track for you! It is,

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use 

All you need to do is, 
1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add you tasks.
4. let it manage your tasks for you! 🥳

And it is **FREE**!

Features:
- [X] Managing Todos
- [X] Managing Deadlines 
- [X] Managing Events
- [ ] Reminders (coming soon...)

If you are a Java programmer, you can use it to practice Java too. Here's the main method: 

```java
public class Main {
    public static void main(String[] args) {
        new Duke("./tasks.txt").run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

peienlim committed:
Implement level 0 - Rename, Greet, Exit

peienlim committed:
Implement level 1 - Echo

peienlim committed:
Implement level 2 - Add, list

peienlim committed:
Implement level 3 - Mark as Done

peienlim committed:
Implement level 4 - ToDos, Events, Deadlines

peienlim committed:
Implement level 4 - ToDos, Events, Deadlines

peienlim committed:
Implement level 5 - A-TextUiTesting

Implement A-TextUiTesting

peienlim committed:
Implement level 5 - handle errors

peienlim committed:
Implement level 6 - delete

peienlim committed:
Implement Level 7 - Save

peienlim committed:
Remove unwanted files

peienlim committed:
Revert "Remove unwanted files"

This reverts commit 31378cf75fed93f4a31d0b8f45add7f551a970ff.

peienlim committed:
Implement level 8 - dates and times

peienlim committed:
Implement level A-MoreOOP

peienlim committed:
Implement Level A-Packages

peienlim committed:
Merge branch 'add-gradle-support'

peienlim committed:
Implement Level A-JUnit

peienlim committed:
Add javadoc comments for Parser.java and Storage.java

peienlim committed:
Adjust code to comply with coding standards

peienlim committed:
Implement Level-9 find keyword function

peienlim committed:
Add javadoc comments for all other non-private classes/methods

peienlim committed:
Merge branch branch-A-CodingStandard

peienlim committed:
Fix merge conflicts in TaskList.java

kokerinks commented on src/main/java/duke/ui/Ui.java:
Can tell that considerable effort was put into method names here to make them clear and easy to understand 👍

kokerinks commented on src/main/java/duke/task/Deadline.java:
```protected LocalDate by;```

perhaps instead of ```by```, can use ```byDate```?

peienlim committed:
Implement JavaFX up to iteration 1

kokerinks commented on src/test/java/duke/task/TaskListTest.java:
More of a coding standard violation, but note that test functions should be named by a specific standard as stated in the [Java Coding Standards](https://se-education.org/guides/conventions/java/intermediate.html):

>Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()
>
>e.g. sortList_emptyList_exceptionThrown() getMember_memberNotFound_nullReturned
>
>Third part or both second and third parts can be omitted depending on what's covered in the test. For example, the test method sortList_emptyList() will test sortList() method for all variations of the 'empty list' scenario and the test method sortList() will test the sortList() method for all scenarios.

kokerinks commented on src/test/java/duke/task/TaskTest.java:
Same as stated in tasklist test:

>Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()
>
>e.g. sortList_emptyList_exceptionThrown() getMember_memberNotFound_nullReturned
>
>Third part or both second and third parts can be omitted depending on what's covered in the test. For example, the test method sortList_emptyList() will test sortList() method for all variations of the 'empty list' scenario and the test method sortList() will test the sortList() method for all scenarios.

kokerinks reviewed (commented):
LGTM! just a few minor issues to look out for

baskargopinath commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,125 @@
+     */
+    public static Object[] parseCommand(String input) throws DukeException {
+        if (input.equals("bye")) {
+            return new Object[] { "bye" };
```
Could a switch statement help instead of if else?

baskargopinath commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,44 @@
+     * @param toDate The end date of the event task.
+     * @param toTime The time at which the event task ends.
+     */
+    public Event(String description, LocalDate fromDate, String fromTime, LocalDate toDate, String toTime) {
```
I like the use of good variable names

baskargopinath commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,104 @@
+    /**
+     * Prints the list of tasks.
+     *
+     * @param tasks An array of tasks to be displayed.
```
Good following of the JavaDocs method description guide

baskargopinath commented on README.md:
```diff
@@ -13,7 +13,7 @@ Prerequisites: JDK 11, update Intellij to the most recent version.
 1. Configure the project to use **JDK 11** (not other versions) as explained in [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).<br>
    In the same dialog, set the **Project language level** field to the `SDK default` option.
-3. After that, locate the `src/main/java/Duke.java` file, right-click it, and choose `Run Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
+3. After that, locate the `src/main/java/duke.Duke.java` file, right-click it, and choose `Run duke.Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
```
Could this be summarized furthur?

baskargopinath commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,19 @@
+Now you have 1 tasks in the list.
+mark 1
+Nice! I've marked this task as done:
+[T][X] borrow book
```
Functionality looks good

baskargopinath reviewed (commented):
LGTM! I appreciate the effort to name methods and variables appropriately

peienlim committed:
Implement Level-10 GUI

peienlim committed:
Add new and edit existing javadoc comments

Dexter-Wong commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,84 @@
+     * @param img    The image to be displayed.
+     * @param isUser A boolean indicating if the speaker is the user.
+     */
+    private DialogBox(String text, Image img, boolean isUser) {
```
Nice naming of methods

Dexter-Wong commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,40 @@
+ * Represents a command to add a deadline task to the task list.
+ */
+public class DeadlineCommand extends Command {
+    private String description;
```
Spacing here is not consistant

Dexter-Wong commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,49 @@
+package duke.command;
+
+import duke.task.Event;
```
Good separation of imports

Dexter-Wong commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,55 @@
+ * Represents a generic task.
+ */
+public class Task {
+    protected String description;
```
Spacing needed

Dexter-Wong reviewed (commented):
Nice work, some spacing issues but is all good.

peienlim committed:
Add assertions to improve reliability

Assertions added to Parser class to ensure validity of task number inputted by user.
Assertions added to Storage class to ensure file path validity.
Added assertions for Deadline and Event classes to check for the validity of time inputs for the tasks.

peienlim committed:
Refactor task package classes for readability

Return statements for toString and toFileString methods very long and difficult to read.

Extracting different parts of the return string into different variables which are named intuitively improve readability of the return string and understanding of the code.

peienlim committed:
Different levels of abstraction for methods in DialogBox and MainWindow class

Should be changed so as to abide with SLAP and improve readability of code.

Abstracted the setting of box alignment code into a new method in DialogBox class.
Code for displaying dialog boxes also abstracted into a new method in MainWindow class.

This ensures that the level of abstractions are the same, abiding with SLAP.

peienlim committed:
Minor refactoring for Storage and Parser classes

peienlim committed:
Merge pull request #1 from peienlim/branch-A-Assertions

Add assertions to improve reliability

peienlim committed:
Merge branch 'master' into branch-A-CodeQuality

peienlim committed:
Merge pull request #2 from peienlim/branch-A-CodeQuality

Refactor code to improve code quality

peienlim committed:
Add tagging for task command and task classes

peienlim committed:
Implement functionality for users to display tagged tasks

peienlim committed:
Fix bug: printing of null as tag for untagged tasks

peienlim committed:
Update Javadoc comments and minor fixes to improve code quality

peienlim committed:
Adjust format of messages sent by cookie

peienlim committed:
Change colours of chatbot components

peienlim committed:
Rename product from Duke to Cookie

peienlim committed:
Rename product from Duke to Cookie

peienlim committed:
Add Ui.png and readme

peienlim committed:
Minor fix for readme and build.gradle

peienlim committed:
Add title for GUI and update Ui.png
